sections/email-signup-banner.liquid

```diff
@@ -0,0 +1,262 @@
+{{ 'section-image-banner.css' | asset_url | stylesheet_tag }}
+{{ 'component-newsletter.css' | asset_url | stylesheet_tag }}
+{{ 'newsletter-section.css' | asset_url | stylesheet_tag }}
+{{ 'section-email-signup-banner.css' | asset_url | stylesheet_tag }}
+{{ 'component-rte.css' | asset_url | stylesheet_tag }}
+
+{%- style -%}
+  #Banner-{{ section.id }}::after {
+    opacity: {{ section.settings.image_overlay_opacity | divided_by: 100.0 }};
+  }
+  {%- if section.settings.adapt_height_image and section.settings.image != blank -%}
+    @media screen and (max-width: 749px) {
+      #Banner-{{ section.id }}:not(.banner--mobile-bottom) .banner__content::before {
+        padding-bottom: {{ 1 | divided_by: section.settings.image.aspect_ratio | times: 100 }}%;
+        content: '';
+        display: block;
+      }
+    }
+
+    #Banner-{{ section.id }}::before,
+    #Banner-{{ section.id }} .banner__media::before {
+      padding-bottom: {{ 1 | divided_by: section.settings.image.aspect_ratio | times: 100 }}%;
+      content: '';
+      display: block;
+    }
+  {%- endif -%}
+{%- endstyle -%}
+
+<div id="Banner-{{ section.id }}" class="email-signup-banner banner banner--medium{% if section.settings.image != blank %} banner--stacked{% endif %}{% if section.settings.adapt_height_image and section.settings.image != blank %} banner--adapt{% endif %}{% if section.settings.show_text_below %} banner--mobile-bottom{%- endif -%}{% if section.settings.show_text_box == false %} banner--desktop-transparent{% endif %}">
```

Comment:
Have we considered offering this type of flexibility to manage the height?

cc. @wiktoriaswiecicka @nicklepine

Reply:
@melissaperreault @wiktoriaswiecicka @nicklepine I am not sure if that is the reason, but thinking about it I think this section is already a combination of banner + email sign up so it already has more settings. Screenshot: https://screenshot.click/13-23-hl7ry-5d3c2.png

So I think that adding more settings, may become a bit much? 🤔 but not sure if that was the reason behind it.

We also have the checkbox `adapt to image height` so the merchant could choose between the two

Reply:
@melissaperreault To play devil's advocate, you're right 🤔. we dont know what kind of image they are uploading so more control does sound like it would be beneficial. I wonder if we have any data on how they are currently using the banner section? 

I am curious if it was a consideration as well

Reply:
This is the main reason why we introduced Image height setting to have more flexibility to accommodate various needs. https://github.com/Shopify/dawn/issues/552 solved by https://github.com/Shopify/dawn/pull/695, we can talk about the risk of providing or not for this particular template. Perhaps I am missing some expectations from that template.

Reply:
Yea 🤔 I can see it going both ways and it can be done as a fast follow as well.

Something to keep in mind is that this section is only available on the password page and it's the combination with the email sign up, so maybe the expectations are a bit different? We can discuss what makes the most sense